### ISSUE TITLE
iojs installation error, even when disabled

### ISSUE THREAD

**Issue by Synchro** [fubarhouse/fubarhouse.npm#22]
This is my nodejs installation playbook:

```
---
- hosts: nodejs
  become: yes
  roles:
    - fubarhouse.nodejs
  vars:
    node_version: 6.3.1
    node_versions:
      - 6.3.1
    install_iojs: false
    install_ivm: false
```

Despite disabling iojs installation, it still tries, and then fails with this error:

```
TASK [fubarhouse.nodejs : IOJS | Get versions] ***********************************************************************************************************************************************************************************************************************
ok: [node1.example.co.uk]

TASK [fubarhouse.nodejs : IOJS | Check for local default install] ****************************************************************************************************************************************************************************************************
skipping: [node1.example.co.uk]

TASK [fubarhouse.nodejs : IOJS | Check for all local installs] *******************************************************************************************************************************************************************************************************
skipping: [node1.example.co.uk] => (item=6.3.1) 

TASK [fubarhouse.nodejs : IOJS | Install default] ********************************************************************************************************************************************************************************************************************
fatal: [node1.example.co.uk]: FAILED! => {"failed": true, "msg": "The conditional check 'fubarhouse_npm_io_install_result is defined and not fubarhouse_npm_io_install_result.stat.exists and \"{{ node_version }}\" in iojs_available_versions.stdout' failed. The error was: error while evaluating conditional (fubarhouse_npm_io_install_result is defined and not fubarhouse_npm_io_install_result.stat.exists and \"{{ node_version }}\" in iojs_available_versions.stdout): 'dict object' has no attribute 'stat'\n\nThe error appears to have been in '/usr/local/etc/ansible/roles/fubarhouse.nodejs/tasks/iojs.yml': line 38, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: \"IOJS | Install default\"\n  ^ here\n"}
```

I made it run by commenting out the lines in main.yml that include iojs.yml, but I don't see why it fails nor why setting those vars doesn't prevent it from trying.

**Comment by fubarhouse** [comment]
@Synchro 

1. Are you running ansible via ssh (vagrant provision for example)?
2. What is the server architecture/version?
3. What version of this role are you using? It might simply pay to update it first as there were some pretty big changes a while ago now which migrated IOJS tasks into a new IVM (iojs version manager) tasks file.
4. If you are using Vagrant, could you please provide a Vagrantfile?

**Comment by Synchro** [comment]
1. yes, remotely, not using vagrant.
2. targets are all Ubuntu 16.04
3. I'm using the role ansible galaxy installed by default - I assume it picked up whatever the current release was 2 days ago.

**Comment by fubarhouse** [comment]
@Synchro thanks, I'll sort this out tonight.

**Comment by fubarhouse** [comment]
@Synchro I've duplicated the environment and your failing task is skipping.

I've still got to look at the conditional logic of that task, but I'm curious as to what version of Ansible is running?

**Comment by Synchro** [comment]
I am running 2.3.0 via homebrew on OS X - I was forced to upgrade to HEAD due to a bug in the current ansible 2.2 release. I guess that could be a problem - but suggests that it may need fixing for 2.3 anyway!

**Comment by fubarhouse** [comment]
Are you able to provide a --verbose output? I will have enough information to put something in place to fix it if you can - the more verbose the better.

**Comment by fubarhouse** [comment]
I've been able to replicate this problem, so it has once again sprung onto my radar.

**Comment by zfogg** [comment]
+1

**Comment by fubarhouse** [comment]
See commit d63a1bd94336281eb6f57f2a1267487b82ff252a for testing.

It currently affects the master branch, and I've visually verified the output of both routes of configuration and I'm happy. The errors caused during execution are resolved, but of course feedback is always appreciated.

I'll give it a couple of days and I'll update the default versions and release this - as it's rather important to get out sooner rather than later.

**Comment by fubarhouse** [comment]
Additionally the task files are now not included, variable references are fixed with defaults set to false for ivm and iojs installations.

**Comment by fubarhouse** [comment]
I'm rather content with this now, but I have separated them out more, identified and fixed a validation issue.

I've also done independent testing for IVM/IOJS and NodeJS to make sure the execution path matches to expectation, and updated the default version number to 7.8.0.

Still awaiting responses on this, so I'll move onto fixing up some stuff in my golang role.

I'll wait exactly 2 days to receive feedback and I'll create a new release and sync up galaxy.

**Comment by Synchro** [comment]
I can't test this easily at the moment, but given you managed to reproduce it, it sounds good to me.

**Comment by fubarhouse** [comment]
Very well, I'll line a release up now :)

**Comment by fubarhouse** [comment]
Fixed in release version 1.6.0.

Feel free to reopen this issue.
